### PR TITLE
[bitnami/thanos] Add Thanos Query Frontend

### LINKED ISSUE #3747: [Bitnami/Thanos] Feature request of Query Frontend
**Which chart**:
Bitnami/Thanos [latest]

**Is your feature request related to a problem? Please describe.**
After metrics observation for our Thanos installation, we noticed latency in Thanos Querier.
This latency comes from long-range queries, during streaming series from Prometheus.

**Describe the solution you'd like**
Add the option in Thanos Chart to install [Query Frontend](https://thanos.io/v0.15/components/query-frontend/#query-frontend) component.

Thank you!

### PR DESCRIPTION
**Pull request by Reuuke**
**Description of the change**

Add ability to deploy [Thanos Query Frontend](https://thanos.io/v0.15/components/query-frontend/#query-frontend) component.

**Benefits**

Users can cache responses for query range.
Users can split query range queries.
Users can rate limit and retry range queries.

**Possible drawbacks**

**Applicable issues**

  - fixes #3747 

**Additional information**

**Checklist** 
- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [X] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

**Comment by andresbono** [comment]
Thanks for opening this PR @Reuuke!! We are going to review it and we will get back to you.

**Review by andresbono — COMMENTED** [review]
Could you also resolve the conflict in `Chart.yaml`?

**Comment by andresbono** [comment]
Thanks you very much for this amazing PR, @Reuuke!! We will still give another round of review before it is accepted and merged. Could you also explain to us if you have tested the changes under some scenarios?

File: bitnami/thanos/templates/query-frontend/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+      {{- end }}
+      containers:
+        - name: query-frontend
+          image: {{ include "thanos.image" . }}
+          imagePullPolicy: {{ .Values.image.pullPolicy | quote }}
+          args:
+            - query-frontend
+            - --log.level={{ .Values.queryFrontend.logLevel }}
+            - --http-address=0.0.0.0:10902
+            - --query-frontend.downstream-url=http://{{ include "thanos.fullname" . }}-querier:9090
```
[the remark this answers is not in the document]
Comment (Reuuke) [reply bitnami/thanos/templates/query-frontend/deployment.yaml]: Fixed.

**Comment by Reuuke** [comment]
> Thank you very much for this amazing PR, @Reuuke!! We will still give another round of review before it is accepted and merged. Could you also explain to us if you have tested the changes under some scenarios?

I have deployed this in our two EKS clusters and verified that it works through Grafana + checked logs that there are no errors.
The performance test did not perform.

**Review by javsalgar — APPROVED** [review]
LGTM

**Comment by bhanugarg23** [comment]
Can anyone help me to expose Thanos query frontend via ingress. I am using nginx ingress controller. Below is the ingress manifest generated from the helm chart 

apiVersion: extensions/v1beta1                                                                                                                                          
kind: Ingress                                                                                                                                                           
metadata:                                                                                                                                                               
  annotations:                                                                                                                     
    nginx.ingress.kubernetes.io/rewrite-target: /$2                                                                                                                     
    nginx.ingress.kubernetes.io/ssl-redirect: "false"                                                                                                                   
    nginx.ingress.kubernetes.io/use-regex: "true"                                                                                                                                                                                                                                             
spec:                                                                                                                                                                   
  ingressClassName: nginx                                                                                                                                               
  rules:                                                                                                                                                                
  - host: myhostname.com                                                                                                                                       
    http:                                                                                                                                                               
      paths:                                                                                                                                                            
      - backend:                                                                                                                                                        
          serviceName: thanos-query-frontend                                                                                                                            
          servicePort: http                                                                                                                                             
        path: /thanos(/|$)(.*)                                                                                                                                          
        pathType: ImplementationSpecific
       
Above implementation is giving me a 404

**Pull request merged**